packages/core/test/financial-templates/common/financial-product-libraries/long-short-pair-libraries/SuccessTokenLongShortPairFinancialProductLibrary.js

```diff
@@ -0,0 +1,141 @@
+const hre = require("hardhat");
+const { runDefaultFixture } = require("@uma/common");
+const { getContract } = hre;
+const { didContractThrow, ZERO_ADDRESS } = require("@uma/common");
+const { assert } = require("chai");
+
+// Tested Contract
+const SuccessTokenLongShortPairFinancialProductLibrary = getContract(
+  "SuccessTokenLongShortPairFinancialProductLibrary"
+);
+
+const LongShortPairMock = getContract("LongShortPairMock");
+
+const { toWei, toBN } = web3.utils;
+const strikePrice = toWei("400");
+const basePercentage1 = toBN(toWei("0.5"));
+const basePercentage2 = toBN(toWei("0.2"));
+const collateralPerPair = toBN(toWei("1"));
+
+describe("SuccessTokenLongShortPairFinancialProductLibrary", function () {
+  let successTokenLSPFPL;
+  let lspMock;
+  let accounts;
+
+  before(async () => {
+    await runDefaultFixture(hre);
+    accounts = await hre.web3.eth.getAccounts();
+  });
+
+  beforeEach(async () => {
+    successTokenLSPFPL = await SuccessTokenLongShortPairFinancialProductLibrary.new().send({ from: accounts[0] });
+    lspMock = await LongShortPairMock.new(
+      "1000000", // _expirationTimestamp
+      collateralPerPair // _collateralPerPair
+    ).send({ from: accounts[0] });
+  });
+  describe("Long Short Pair Parameterization", () => {
+    it("Can set and fetch valid strikes", async () => {
+      await successTokenLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, strikePrice, basePercentage1)
+        .send({ from: accounts[0] });
+
+      const setParams = await successTokenLSPFPL.methods.longShortPairParameters(lspMock.options.address).call();
+      assert.equal(setParams.strikePrice.toString(), strikePrice);
+    });
+    it("Can not re-use existing LSP contract address", async () => {
+      await successTokenLSPFPL.methods
+        .setLongShortPairParameters(lspMock.options.address, strikePrice, basePercentage1)
+        .send({ from: accounts[0] });
+
+      // Second attempt should revert.
+      assert(
+        await didContractThrow(
+          successTokenLSPFPL.methods
+            .setLongShortPairParameters(lspMock.options.address, strikePrice, basePercentage1)
+            .send({ from: accounts[0] })
+        )
+      );
+    });
+    it("Can not set invalid LSP contract address", async () => {
+      // LSP Address must implement the `expirationTimestamp method.
+      assert(
+        await didContractThrow(
+          successTokenLSPFPL.methods
+            .setLongShortPairParameters(ZERO_ADDRESS, strikePrice, basePercentage1)
+            .send({ from: accounts[0] })
+        )
+      );
+    });
+  });
+  describe("Compute expiry tokens for collateral with base percentage of 0.5", () => {
```

Comment:
Can you add a test for when the expiry price = 0 since this is an edge case in the contract?